mitodl/mit-open
# Pace and format fields for LearningResource model

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5127
Closes https://github.com/mitodl/hq/issues/5128

### Description (What does it do?)
Adds the `format` and `pace` fields and populates them via ETL pipelines


### How can this be tested?
- Copy all the .env values needed for ETL pipelines (see `settings_course_etl.py`) from RC, except contentfile-related settings (buckets for contentfiles).
- Run `docker compose up` or `docker compose run --rm web ./manage.py migrate`.  The new migration should complete successfully.
- Run all the following pipelines, they should complete successfully:
  ```
  ./manage.py backpopulate_mitx_edx_data
  ./manage.py backpopulate_micromasters_data
  ./manage.py backpopulate_mitxonline_data
  ./backpopulate_ocw_data --course-name 21w-758-genre-fiction-workshop-fantasy-fall-2016 --skip-contentfiles
  ./manage.py backpopulate_oll_data
  ./manage.py backpopulate_prolearn_data
  ./manage.py backpopulate_sloan_data
  ./manage.py backpopulate_xpro_data
  ```
- Check that the program "Supply Chain Management" has a `pace` value of `["self_paced", "instructor_paced"]`

### Additional Context
Pace and format were modeled as lists at the resource level because it's possible that a program might have multiple values depending on the courses that make up the program (some might be a mix of sync and async, instructor-paced and self-paced).
Also, at one point there was a Sloan course with runs that were a mix of instructor-paced and self-paced, sync and async (though that is no longer the case).
